src/figuresClasses.ts

```diff
@@ -1,11 +1,92 @@
-export interface Figure {}
+type Shape = 'triangle' | 'circle' | 'rectangle';
+type Color = 'red' | 'green' | 'blue';
 
-export class Triangle implements Figure {}
+export interface Figure {
+  shape: Shape;
+  color: Color;
+  getArea: () => number;
+}
+
+function checkOnError(errorMessage: string, ...args: number[]): void {
+  if (args.some((arg: number) => arg <= 0)) {
+    throw new Error(errorMessage);
+  }
+}
+
+export class Triangle implements Figure {
+  constructor(
+    public color: Color,
+    public a: number,
+    public b: number,
+    public c: number,
+    public shape: Shape = 'triangle',
+  ) {
+    if (a <= 0 || b <= 0 || c <= 0) {
+      throw new Error(
+        `Triangle sides ${a}, ${b} and ${c} must be greater than zero`,
+      );
+    }
```

Comment:
The checkOnError function checks whether all arguments are greater than zero and throws an error if not. For consistency, you can use the function in the Triangle class to check the sides, which will simplify the code. 

Example:

```suggestion
    checkOnError(
      `Triangle sides ${a}, ${b} and ${c} must be greater than zero`,
      a,
      b,
      c,
    );
```